From 98ddbf815110fa164b23b49b0f06624d3a05ae1f Mon Sep 17 00:00:00 2001
From: Weijie Gao <weijie.gao@mediatek.com>
Date: Thu, 27 Mar 2025 09:59:32 +0800
Subject: [PATCH] scripts/mkits.sh: use "data-padding-alignment" for rootfs
 padding

Use "data-padding-alignment" for rootfs padding instead of "dd" to avoid a
issue that multiple mkits.sh be called in parallel which causes
rootfs.*.pagesync being corrupted while mkimage is calling dtc to generate
fitblk image.

Signed-off-by: Weijie Gao <weijie.gao@mediatek.com>
---
 scripts/mkits.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/mkits.sh b/scripts/mkits.sh
index 46ab5ee023..aed8c935b9 100755
--- a/scripts/mkits.sh
+++ b/scripts/mkits.sh
@@ -136,12 +136,12 @@ fi
 
 
 if [ -n "${ROOTFS}" ]; then
-	dd if="${ROOTFS}" of="${ROOTFS}.pagesync" bs=4096 conv=sync
 	ROOTFS_NODE="
 		rootfs${REFERENCE_CHAR}$ROOTFSNUM {
 			description = \"${ARCH_UPPER} OpenWrt ${DEVICE} rootfs\";
 			${COMPATIBLE_PROP}
-			data = /incbin/(\"${ROOTFS}.pagesync\");
+			data = /incbin/(\"${ROOTFS}\");
+			data-padding-alignment = <4096>;
 			type = \"filesystem\";
 			arch = \"${ARCH}\";
 			compression = \"none\";
-- 
2.45.2

